src/lib/connect/ws.ts

```diff
@@ -220,41 +224,27 @@ const browserStreamBuilder: StreamBuilder = (client, opts) => {
 	}
 
 	function onOpen() {
-		stream.setReadable(proxy)
-		stream.setWritable(proxy)
-		stream.emit('connect')
+		if (stream instanceof BufferedDuplex) {
+			stream.onSocketOpen()
+		}
 	}
 
 	function onClose() {
 		stream.end()
 		stream.destroy()
 	}
 
-	function onError(err: Event) {
-		stream.destroy(err)
+	function onError(err: ErrorEvent) {
+		stream.destroy(err as any)
 	}
 
 	function onMessage(event: MessageEvent) {
 		let { data } = event
 		if (data instanceof ArrayBuffer) data = Buffer.from(data)
-		else data = Buffer.from(data, 'utf8')
+		else data = Buffer.from(data as string, 'utf8')
 		proxy.push(data)
 	}
 
-	// this is to be enabled only if objectMode is false
-	function writev(chunks: any, cb: (err?: Error) => void) {
-		const buffers = new Array(chunks.length)
-		for (let i = 0; i < chunks.length; i++) {
-			if (typeof chunks[i].chunk === 'string') {
-				buffers[i] = Buffer.from(chunks[i], 'utf8')
-			} else {
-				buffers[i] = chunks[i].chunk
-			}
-		}
-
-		this._write(Buffer.concat(buffers), 'binary', cb)
-	}
-
 	function socketWriteBrowser(
```

Comment:
Wondering if there should be a return after setTimeout [here](https://github.com/mqttjs/MQTT.js/pull/1732/files#diff-cc6fb47ee82c471804f56a0629299e3e03381c08ce37d2b3053c259badf7fa64L265) ?